src/bp/core/repositories/conversations.ts

```diff
@@ -0,0 +1,190 @@
+import * as sdk from 'botpress/sdk'
+import { inject, injectable, postConstruct } from 'inversify'
+
+import Database from '../database'
+import { TYPES } from '../types'
+import LRU from 'lru-cache'
+import ms from 'ms'
+import { MessageRepository } from './messages'
+import { JobService } from 'core/services/job-service'
+
+export interface ConversationRepository {
+  getAll(endpoint: sdk.UserEndpoint): Promise<sdk.Conversation[]>
+  getAllRecent(endpoint: sdk.UserEndpoint, limit?: number): Promise<sdk.RecentConversation[]>
+  deleteAll(endpoint: sdk.UserEndpoint): Promise<number>
+  create(endpoint: sdk.UserEndpoint): Promise<sdk.Conversation>
+  getMostRecent(endpoint: sdk.UserEndpoint): Promise<sdk.Conversation | undefined>
+  getById(conversationId: number): Promise<sdk.Conversation | undefined>
+  delete(conversationId: number): Promise<boolean>
+  query()
+  serialize(conversation: Partial<sdk.Conversation>)
+  deserialize(conversation: any)
+}
+
+@injectable()
+export class KnexConversationRepository implements ConversationRepository {
+  private readonly TABLE_NAME = 'conversations'
+  private cache = new LRU<number, sdk.Conversation>({ max: 10000, maxAge: ms('5min') })
+  private invalidateConvCache: Function = this._localInvalidateConvCache
+
+  constructor(
+    @inject(TYPES.Database) private database: Database,
+    @inject(TYPES.JobService) private jobService: JobService,
+    @inject(TYPES.MessageRepository) private messageRepo: MessageRepository
+  ) {}
+
+  @postConstruct()
+  async init() {
+    this.invalidateConvCache = await this.jobService.broadcast<void>(this._localInvalidateConvCache.bind(this))
+  }
+
+  public async getAll(endpoint: sdk.UserEndpoint): Promise<sdk.Conversation[]> {
```

Comment:
This is meant to be channel neutral. You only need to specify the channel when you are sending a message from a channel module. Otherwise you don't need to worry about channels.